### ISSUE TITLE
Collapsible Items within Isotope 

### ISSUE THREAD

**Issue by TrendyGuy** [metafizzy/isotope#1131]
Hi,

I have a few divs that work perfect with isotope but I also have collapsible text elements that disturb the layout of isotope. I can't figure out a way that after the page loads it triggers a isotope to trigger the layout command making everything fit properly. Also, when I open and close the content how to trigger it as well. 

I know it has been discussed all over in reference to imagesLoaded but my problem is not images. The problem is the collapse plugin I am using loads the elements and then collapses them which throws off the layout. I am using jquery.shorten.js to do this. Any help would be appreciated or maybe you ahve a better collapse program that is more friendly with isotope. 

Thanks!

**Comment by desandro** [comment]
I'm sorry to see you're having trouble with Isotope. Could you provide a **reduced test case**? See [Submitting Issues in the contributing guidelines](https://github.com/metafizzy/isotope/blob/master/.github/contributing.md#submitting-issues).

**Comment by TrendyGuy** [comment]
Sorry about that. Here is an example of my problem: http://codepen.io/TrendyGuy/pen/PzKOor

It does not redo the layout after load or after clicking the more button to give everything proper spacing.

Also, I would like to try and default sorting to by "name" instead of original order but I am unsure how to do that. 

Thanks for any help!

**Comment by phongtran0** [comment]
Hi,

Don't mind me, saw the issue and had some time.

`// init Isotope
var $grid = $('.grid').isotope({
  itemSelector: '.element-item',
  layoutMode: 'fitRows',
  sortBy: 'name',
  getSortData: {
    name: '.name',
    symbol: '.symbol',
    number: '.number parseInt',
    category: '[data-category]',
    weight: function( itemElem ) {
      var weight = $( itemElem ).find('.weight').text();
      return parseFloat( weight.replace( /[\(\)]/g, '') );
    }
  }
});

$(document).ready(function() {
	
  $(".more").shorten({
     showChars: 10,
    onMore: function() {
      $('.grid').isotope('layout');  
    },
    onLess: function() {
      $('.grid').isotope('layout');  
    }
  });`

Basically the sortBy: 'name' parameter is pretty straight forward.  When using shorten, they provide onMore and onLess callback functions that you can use to trigger the isotope layout.  Is that the behavior you're trying to do?

**Comment by TrendyGuy** [comment]
Thanks for that it worked perfect. Thanks!

**Comment by desandro** [comment]
Thank you @phongtran0 !
